### PR TITLE
STAR-Fusion wrapper

### PR DESCRIPTION
**Pull request by yhoogstrate**
Another wrapper that I couldn't get working because of the dependencies. I tested this with the conda dependency (https://github.com/bioconda/bioconda-recipes/pull/1190) and that worked.

Also a few minor updates to RNA-STAR

Todo's:
 - [x] Check whether this runs with a reference from history
 - [x] Decide whether or not to use `<tool id="..." profile="16.07">` because of conda dependency

### CODE REVIEW

**Comment by bgruening** [comment]
Awesome! :+1:

**Comment by yhoogstrate** [comment]
I've set this PR to WIP because it can only be installed using bioconda and not with the classical tool shed. Should we create a milestone for anything like this?

**Comment by bgruening** [comment]
@yhoogstrate yes please do so ... I'm all for it ;)

File: tools/star_fusion/star_fusion.xml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        </conditional>
+        
+        <conditional name="params">
+            <param name="settingsType" type="select" label="Settings to use" help="You can use the default settings or set custom values for any STAR Fusion parameter.">
+                <option value="default" selected="true">Use Defaults</option>
+                <option value="full">Full parameter list</option>
+            </param>
+            <when value="default" />
+            <when value="full"><!-- Full/advanced params. -->
+                <param name="min_junction_reads" type="integer" value="1" label="minimum number of junction-spanning reads required." help="(--min_junction_reads)" />
```
Comment (shiltemann) [inline tools/star_fusion/star_fusion.xml]: you could use `argument="--min_junction_reads"` here [right](https://wiki.galaxyproject.org/Admin/Tools/ToolConfigSyntax#A.3Cparam.3E_tag_set)?

File: tools/star_fusion/star_fusion.xml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                <param name="min_novel_junction_support" type="integer" value="3" label="minimum of 3 junction reads required if breakpoint lacks involvement of only reference junctions" help="(--min_novel_junction_support)" />
+                <param name="min_alt_pct_junction" type="float" value="10" label="10% of the dominant isoform junction support" help="(--min_alt_pct_junction)" />
+                <param name="aggregate_novel_junction_dist" type="integer" value="5" label="non-ref junctions within 5 are merged into single calls" help="(--aggregate_novel_junction_dist)" />
+                <param name="E" type="float" value="0.001" label="E-value threshold for blast searches" help="(-E)" />
+            </when>
+        </conditional>
+    </inputs>
+
+    <outputs>
+        <data format="tabular" name="output_final" label=".fusion_candidates.final" from_work_dir="star-fusion.fusion_candidates.final"/>
```
Comment (shiltemann) [inline tools/star_fusion/star_fusion.xml]: `${tool.name} on ${on_string}: fusion_candidates.final` ?

File: tools/star_fusion/star_fusion.xml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        <param name="chimeric_junction" type="data" format="interval" label="Chimeric junction file from STAR (with STAR-Fusion settings)" />
+        
+        <!-- Genome source. -->
+        <conditional name="refGenomeSource">
+            <param name="genomeSource" type="select" label="Custom or built-in reference genome" help="Built-ins were indexed using default options">
+                <option value="indexed" selected="True">Use a built-in index</option>
+                <option value="history">Index and use a genome fasta and GTF file from history</option>
+            </param>
+            <when value="indexed">
+            <param name="index" type="select" label="Select a reference genome">
```
Comment (shiltemann) [inline tools/star_fusion/star_fusion.xml]: indent this block

**Comment by bgruening** [comment]
@yhoogstrate let's merge this as soon as you updated the latest version! Thanks!

**Comment by bgruening** [comment]
Awesome!

**Pull request merged**